File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(str_a, str_b)
    count = 0
    pos = 0
    raise ArgumentError if str_a.length != str_b.length
    while pos < str_a.length
      count += 1 if str_a[pos] != str_b[pos]
      pos += 1
    end
    count
  end
end

module BookKeeping
  VERSION = 3
end

File: javascript/hamming/hamming.js
function Hamming() {}

Hamming.prototype.compute = function doCompute(strA, strB) {
  if (strA.length !== strB.length) {
    throw Error('DNA strands must be of equal length.');
  }
  const distance = strA.split('').filter((char, index) => char !== strB[index]);
  return distance.length;
};


module.exports = Hamming;
